transcom/mymove
# B 18634 int

## [B-18634-INT](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18634)

## Summary

This task allows Service Counselors to view the MTO Page already available to TXO roles. However, it needs to be displayed as a read-only page, and so all interact-able elements on the page have been restricted with an additional permission type that Service Counselor roles do not receive, but TXO roles do. This means that all buttons do not render for Service Counselors, making the page read-only. 

This task is a downstream of backlogs B-18670 and B-18909 due to the technical implementation.

Please note that this cannot be tested with a multi-role office user as they receive all permissions available, even when appearing as the Service Counselor role.

Is there anything you would like reviewers to give additional scrutiny?
Please ensure that there are no interact-able elements that are missing this new permissions restriction during any part of the move process. Additionally, please ensure all frontend tests are passing.

Explain more about the approach used.
Originally, this task was completed by implementing role checks on every single sub-component of the MTO page, but since this required reading the state for each individual page, it was decided that creating a new permission and nesting it within existing permissions for all features was the most streamlined approach to solving the issue. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app and sign in as a Services Counselor (NOT MULTI ROLE)
2. Find appropriate moves to view the details of. AMDORD is a good example, there should be enough to test in our auto-generated test data.
3. When you select the move, it will bring you to the Move Details page. Please locate the "Move task order" tab next to the Move Details tab.
4. Click the "Move task order" tab.
5. Ensure the page routes correctly to the MTO page with counseling in the url bar, it retains the Service Counselor Nav Bar, contains all necessary information, and contains no interact-able elements or buttons. 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
Example of functional read-only MTO page for Service Counselor
<img width="1571" alt="image" src="https://github.com/transcom/mymove/assets/148906585/9c4f3afe-387d-4e09-8701-3ca21e206e3a">
